runtime/src/bank.rs

```diff
@@ -4676,10 +4676,7 @@ impl Bank {
     /// Process a Transaction. This is used for unit tests and simply calls the vector
     /// Bank::process_transactions method.
     pub fn process_transaction(&self, tx: &Transaction) -> Result<()> {
-        self.try_process_transactions(std::iter::once(tx))?[0].clone()?;
-        tx.signatures
-            .first()
-            .map_or(Ok(()), |sig| self.get_signature_status(sig).unwrap())
+        self.try_process_transactions(std::iter::once(tx))?[0].clone()
```

Comment:
Had to dig a little, but this looks like a reasonable simplification:
- `get_signature_status()` ends up reading the status back from status_cache via `get_signature_status_slot()`
- `try_process_transactions()` eventually calls `update_transaction_statuses()` which updates status_cache

So agreed that is it valid to remove the status_cache lookup when we have status already available